src/components/molecules/Navbar.tsx

```diff
@@ -18,30 +20,42 @@ const Elements = styled(Container)({
   alignItems: "center",
 });
 
-const StyledLink = styled(Link)(({theme}) =>  ({
-  textDecoration: 'none',
-  color: 'inherit',
-  '&:hover': {
-    color: theme.palette.secondary.light
-  }
+const StyledLink = styled(Link)(({ theme }) => ({
+  textDecoration: "none",
+  color: "inherit",
+  "&:hover": {
+    color: theme.palette.secondary.main,
+  },
 }));
 
-const Links = styled('div')({
-    display: 'flex',
-    columnGap: '1rem'
-})
+const StyledDiv = styled("div")(({ theme }) => ({
+  cursor: "pointer",
+  "&:hover": {
+    color: theme.palette.secondary.main,
+  },
+}));
+
+const Links = styled("div")({
+  display: "flex",
+  columnGap: "1rem",
+});
 
 export const Navbar = () => {
+  const dispatch = useAppDispatch();
+
+  const handleLogout = () => {
+    dispatch(logout());
```

Comment:
I would use tanstack query and put there `auth.signOut` https://firebase.google.com/docs/auth/web/password-auth?hl=pl